sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -574,6 +574,19 @@ case class RangeKeyScanStateEncoderSpec(
  */
 trait StateStoreProvider {
 
+  // Track whether this state store instance is lagging behind in snapshot uploads.
```

Comment:
Lets not set this as a member of the provider. The provider is long lived across many batches and store instances. To avoid race or bug, where many stores are now mistakenly uploading snapshots. This flag should be pass-through to the specific State store instance created